bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,458 @@
+        // *****getMspConfig from Hayward server
+        String urlParameters = "<?xml version=\"1.0\" encoding=\"utf-8\"?><Request><Name>GetMspConfigFile</Name><Parameters>"
+                + "<Parameter name=\"Token\" dataType=\"String\">" + account.token + "</Parameter>"
+                + "<Parameter name=\"MspSystemID\" dataType=\"int\">" + account.mspSystemID
+                + "</Parameter><Parameter name=\"Version\" dataType=\"string\">0</Parameter>\r\n"
+                + "</Parameters></Request>";
+
+        String xmlResponse = httpXmlResponse(urlParameters);
+
+        // Debug: Inject xml file for testing
+        // String path =
+        // "C:/Users/Controls/openhab-2-5-x/git/openhab-addons/bundles/org.openhab.binding.haywardomnilogic/getConfig.xml";
+        // xmlResponse = new String(Files.readAllBytes(Paths.get(path)));
+
+        if (xmlResponse.isEmpty()) {
+            logger.debug("Hayward Connection thing: requestConfig XML response was null");
+            return "Fail";
+        }
+
+        if (evaluateXPath("//Backyard/Name/text()", xmlResponse).isEmpty()) {
+            logger.debug("Hayward Connection thing: requestConfiguration XML response: {}", xmlResponse);
+            return "Fail";
+        }
+        return xmlResponse;
+    }
+
+    public synchronized boolean getTelemetryData() throws HaywardException {
+        // *****getTelemetry from Hayward server
+        String urlParameters = "<?xml version=\"1.0\" encoding=\"utf-8\"?><Request><Name>GetTelemetryData</Name><Parameters>"
+                + "<Parameter name=\"Token\" dataType=\"String\">" + account.token + "</Parameter>"
+                + "<Parameter name=\"MspSystemID\" dataType=\"int\">" + account.mspSystemID
+                + "</Parameter></Parameters></Request>";
+
+        String xmlResponse = httpXmlResponse(urlParameters);
+
+        if (xmlResponse.isEmpty()) {
+            logger.debug("Hayward Connection thing: getTelemetry XML response was null");
+            return false;
+        }
+
+        if (!evaluateXPath("/Response/Parameters//Parameter[@name='StatusMessage']/text()", xmlResponse).isEmpty()) {
+            logger.debug("Hayward Connection thing: getTelemetry XML response: {}", xmlResponse);
+            return false;
+        }
+
+        for (Thing thing : getThing().getThings()) {
+            if (thing.getHandler() instanceof HaywardThingHandler) {
+                HaywardThingHandler handler = (HaywardThingHandler) thing.getHandler();
+                if (handler != null) {
+                    handler.getTelemetry(xmlResponse);
+                }
+            }
+        }
+        return true;
+    }
+
+    public synchronized boolean getAlarmList() throws Exception {
+        for (Thing thing : getThing().getThings()) {
+            Map<String, String> properties = thing.getProperties();
+            if (properties.get(HaywardBindingConstants.PROPERTY_TYPE).equals("BACKYARD")) {
```

Comment:
If you swap these, you avoid any null error.
```suggestion
            if ("BACKYARD".equals(properties.get(HaywardBindingConstants.PROPERTY_TYPE))) {
```